Multiple primary keys creates unhelpful generated code errors

IHP's UI does little to stop you from creating a table with two primary keys.

Using the UI I was able to make [this schema](https://gist.github.com/DrewFenwick/093a8e2cd7d69c45f2155f5ba49ef3e6),
which caused the following errors to show up in HLS

```hs

I see, well yes maybe another name would be better. Would `pk` work? It's maybe not as obvious what it means though. 

Also, I just ran my project with the latest dev server and I'm getting this: 

```
Mentor/FrontController.hs:29:9: error:
    • Couldn't match type ‘Id' "users"’ with ‘UUID’
        arising from a use of ‘initAuthentication’
    • In a stmt of a 'do' block: initAuthentication @User
      In the expression:
        do setLayout defaultLayout
           initAutoRefresh
           initAuthentication @User
           initUserAdminInfo
      In an equation for ‘initContext’:
          initContext
            = do setLayout defaultLayout
                 initAutoRefresh
                 initAuthentication @User
                 ....
   |
29 |         initAuthentication @User
```

Is this related? Weird thing is that this was working just now, when I submitted my pagination PR. Have you committed since?

I pulled the very latest and now I'm getting spammed with: 

```
build/Generated/Types.hs:89:22: error:
    • No instance for (Default (Id' "users"))
        arising from a use of ‘def’
    • In the first argument of ‘User’, namely ‘def’
      In the expression:
        User def def def Nothing def def def def def def def
      In an equation for ‘newRecord’:
          newRecord = User def def def Nothing def def def def def def def
   |
89 |     newRecord = User def def def Nothing def def def def def def def
``` 

for many types. You probably already know this. Just posting it in case it's helpful.

> I pulled the very latest and now I'm getting spammed with:
> 
> ```
> build/Generated/Types.hs:89:22: error:
>     • No instance for (Default (Id' "users"))
>         arising from a use of ‘def’
>     • In the first argument of ‘User’, namely ‘def’
>       In the expression:
>         User def def def Nothing def def def def def def def
>       In an equation for ‘newRecord’:
>           newRecord = User def def def Nothing def def def def def def def
>    |
> 89 |     newRecord = User def def def Nothing def def def def def def def
> ```
> 
> for many types. You probably already know this. Just posting it in case it's helpful.

This same thing just happened to me on our "production" server, and the only thing that worked was doing this:

> It was weird. The first time I ran `make -B .envrc` it didn't work. What did seem to work was running `main` from `:l ...DevServer.hs` in `ghci` instead of running `main` from `:l Main`. Now `:l Main` also works. 👍

Seems that just `make -B .envrc` wasn't enough.

--- a/ihp-ide/Test/SchemaCompilerSpec.hs
+++ b/ihp-ide/Test/SchemaCompilerSpec.hs
@@ -770,6 +770,53 @@ tests = do
                         {-# INLINE filterWhereId #-}
                     |]
 
+        describe "needsHasFieldId" do
+            let
+                isNamedTable :: Text -> Statement -> Bool
+                isNamedTable targetName (StatementCreateTable CreateTable { name }) = name == targetName
+                isNamedTable _ _ = False
+            it "should not generate HasField id for composite PK table with an id column" do
+                let statements = parseSqlStatements [trimming|
+                    CREATE TABLE ideas_votes (
+                        id INT NOT NULL,
+                        idea_id UUID NOT NULL,
+                        parent_id UUID NOT NULL,
+                        PRIMARY KEY(idea_id, parent_id)
+                    );
+                |]
+                let (Just statement) = find (isNamedTable "ideas_votes") statements
+                let compileOutput = compileStatementPreview statements statement |> Text.strip
+
+                -- Should NOT contain a generated HasField "id" instance since the table has a column named "id"
+                compileOutput `shouldNotSatisfy` (Text.isInfixOf "instance HasField \"id\"")
+
+            it "should not generate HasField id for single non-id PK table with an id column" do
+                let statements = parseSqlStatements [trimming|
+                    CREATE TABLE things (
+                        id INT NOT NULL,
+                        code TEXT PRIMARY KEY NOT NULL
+                    );
+                |]
+                let (Just statement) = find (isNamedTable "things") statements
+                let compileOutput = compileStatementPreview statements statement |> Text.strip
+
+                -- Should NOT contain a generated HasField "id" instance since the table has a column named "id"
+                compileOutput `shouldNotSatisfy` (Text.isInfixOf "instance HasField \"id\"")
+
+            it "should generate HasField id for composite PK table without an id column" do
+                let statements = parseSqlStatements [trimming|
+                    CREATE TABLE bit_part_refs (
+                        bit_ref UUID NOT NULL,
+                        part_ref UUID NOT NULL,
+                        PRIMARY KEY(bit_ref, part_ref)
+                    );
+                |]
+                let (Just statement) = find (isNamedTable "bit_part_refs") statements
+                let compileOutput = compileStatementPreview statements statement |> Text.strip
+
+                -- Should contain a generated HasField "id" instance for the composite PK
+                compileOutput `shouldSatisfy` (Text.isInfixOf "instance HasField \"id\"")
+
         describe "simple mode (compileRelationSupport = False)" do
             let simpleOptions = previewCompilerOptions { compileRelationSupport = False }
             it "should produce no type parameters and no QueryBuilder fields for a table with FK and has-many relations" do

--- a/ihp-schema-compiler/IHP/SchemaCompiler.hs
+++ b/ihp-schema-compiler/IHP/SchemaCompiler.hs
@@ -1144,11 +1144,13 @@ instance HasField "id" #{tableNameToModelName name} (Id' "#{name}") where
             ids -> "Id (" <> commaSep (map columnNameToFieldName ids) <> ")"
 
 needsHasFieldId :: CreateTable -> Bool
-needsHasFieldId CreateTable { primaryKeyConstraint } =
+needsHasFieldId CreateTable { columns, primaryKeyConstraint } =
   case primaryKeyColumnNames primaryKeyConstraint of
     [] -> False
     ["id"] -> False
-    _ -> True
+    pkCols
+      | any (\col -> col.name == "id") columns -> False
+      | otherwise -> True
 
 primaryKeyColumns :: CreateTable -> [Column]
 primaryKeyColumns CreateTable { name, columns, primaryKeyConstraint } =

